### PR TITLE
LOOFAH-38 Merge skills by category and skills by grade queries

### PR DESCRIPTION
**Pull request by elishkaflint**

### CODE REVIEW

**Review by Caitlin-cooling — APPROVED** [review]

File: src/main/java/com/loofah/graph/api/queries/SkillsQuery.java [hunk 1]
```diff
@@ -13,6 +17,7 @@
 public class SkillsQuery implements DataFetcher<List<Skill>> {
 
     private final SkillRepository skillRepository;
+    private final ObjectMapper objectMapper = new ObjectMapper();
```
Comment (Learner5200) [inline src/main/java/com/loofah/graph/api/queries/SkillsQuery.java]: Nitpick: if we made this a bean and injected it into the constructor, we could mock it where necessary in unit tests. e.g. to simulate any errors while converting

File: src/main/java/com/loofah/graph/api/queries/SkillsQuery.java [hunk 2]
```diff
@@ -21,7 +26,27 @@ public SkillsQuery(final SkillRepository skillRepository) {
 
     @Override
     public List<Skill> get(final DataFetchingEnvironment dataFetchingEnvironment) {
-        return skillRepository.findAll();
+
+        SkillFilter skillFilter = objectMapper.convertValue(dataFetchingEnvironment.getArgument("filter"), SkillFilter.class);
+
+        Query query = new Query();
+
+        if(skillFilter != null) {
```
Comment (Learner5200) [inline src/main/java/com/loofah/graph/api/queries/SkillsQuery.java]: Nitpick: as an idea, we could remove some of the nesting in this conditional logic by using optionals. e.g. something like:

```
Optional<SkillFilter> skillFilter = Optional.ofNullable(skillFilter);

skillFilter.map(filter::getCategoryId).ifPresent( {{add the categoryId to the query}} )
skillFilter.map(filter::getGradeId).ifPresent( {{add the gradeId to the query}} )
```

No worries if you find it more readable as is though :)

**Review by Learner5200 — APPROVED** [review]

File: src/main/java/com/loofah/graph/api/queries/SkillsQuery.java (hunk 2 above)
Comment (elishkaflint) [reply src/main/java/com/loofah/graph/api/queries/SkillsQuery.java]: good suggestion, have updated

File: src/main/java/com/loofah/graph/api/queries/SkillsQuery.java (hunk 1 above)
Comment (elishkaflint) [reply src/main/java/com/loofah/graph/api/queries/SkillsQuery.java]: yep makes sense, it's meant i have to add some more when's into the SkillsQueryTest and not sure it's completely necessary - sure there will be a chance to adjust that on a future ticket

**Pull request merged**